R/zchunk_L111.nonghg_en_R_S_T_Y.R

```diff
@@ -1,111 +1,287 @@
+      L111.nonghg_tg_R_en_Si_F_Yh_intl.mlt
+
+    L111.nonghg_tg_R_en_Sedgar_Yh.mlt %>%
+      filter(!(EDGAR_agg_sector %in% c("trn_intl_ship", "trn_intl_air"))) ->
+      L111.nonghg_tg_R_en_Si_F_Yh_dom.mlt
+
+    # Calculate total energy for international aviation emissions
+    L111.nonghg_tg_R_en_Si_F_Yh_intl.mlt %>%
+      group_by(EDGAR_agg_sector, Non.CO2, year) %>%
+      summarise(tot_energy = sum(energy)) %>%
+      ungroup ->
+      L111.en_Si_F_Yh_intl.mlt
+
+    # Compute international input emissions based on total emission and total emission shares
+    L111.nonghg_tg_R_en_Si_F_Yh_intl.mlt %>%
+      left_join(L111.en_Si_F_Yh_intl.mlt, by = c("EDGAR_agg_sector", "Non.CO2", "year")) %>%
+      left_join(L111_EDGAR_intl, by = c("EDGAR_agg_sector", "Non.CO2", "year")) %>%
+      # EDGAR emissions are in Gg, need to convert to Tg
+      mutate(emissions = (value * CONV_GG_TG) * (energy / tot_energy)) %>%
+      select(-tot_energy, -value) %>%
+      na.omit() %>%
+      # Add domestic emissions back in
+      bind_rows(L111.nonghg_tg_R_en_Si_F_Yh_dom.mlt) ->
+      L111.nonghg_tg_R_en_Si_F_Yh.mlt
+
+    # Map in final GCAM sector, technology, and driver type, and calculate energy totals (136-146)
+
+    L111.nonghg_tg_R_en_Si_F_Yh.mlt %>%
+      left_join_keep_first_only(GCAM_sector_tech, by = c("sector", "fuel", "technology")) %>%
+      # Sum emissions by region, non-CO2, sector, technology, and year for EDGAR historical years
+      group_by(GCAM_region_ID, Non.CO2, supplysector, subsector, stub.technology, year) %>%
+      summarise(value = sum(emissions)) %>%
+      ungroup %>%
+      filter(year %in% emissions.EDGAR_YEARS) ->
+      L111.nonghg_tg_R_en_S_F_Yh   # OUTPUT
+
+    # Compute non-ghg emission factors for GCAM sector, fuel, technology, and driver type (148-156)
+
+    L101.in_EJ_R_en_Si_F_Yh %>%
+      gather(year, value, matches(YEAR_PATTERN)) %>%
+      mutate(year = as.integer(year)) %>%
+      left_join_keep_first_only(select(GCAM_sector_tech, sector, fuel, technology, supplysector, subsector, stub.technology),
+                                by = c("sector", "fuel", "technology")) %>%
+      # need to filter here to replicate behavior of original aggregate()
+      filter(!is.na(supplysector), !is.na(subsector), !is.na(stub.technology)) %>%
+      group_by(GCAM_region_ID, supplysector, subsector, stub.technology, year) %>%
+      summarise(energy = sum(value)) %>%
+      ungroup ->
+      L101.in_EJ_R_en_S_F_Yh.mlt
+
+    # Calculate emission shares by energy totals
+
+    L111.nonghg_tg_R_en_S_F_Yh %>%
+      left_join(L101.in_EJ_R_en_S_F_Yh.mlt, by = c("GCAM_region_ID", "supplysector", "subsector", "stub.technology", "year")) %>%
+      # the old data system was doing a straight division of some _tiny_ numbers here, which made it impossible to replicate
+      # we now round; see
+      mutate(value = round(value, 8) / round(energy, 8)) %>%
```

Comment:
You really shouldn't use `round` for this.  "8 decimal places" may be a reasonable level of precision, or it may be far too much or too little, depending on the magnitude of the argument.  With `signif` you always know what the number of places represents.

Reply:
Good point, thanks.